### PR TITLE
pm: fix expected ticket expiration params backwards compatability

### PR DESCRIPTION
**Pull request by kyriediculous**
**What does this pull request do? Explain your changes. (required)**
This PR fixes the backwards compatibility requirement for #1428. 

Originally there is a check that retrieves the necessary `ExpirationParams` from the `Timewatcher` if the Orchestrator does not provide these in `TicketParams`
```
	expirationParams := ticketParams.ExpirationParams
	// Ensure backwards compatbility
	// If no expirationParams are included by O
	// B sets the values based upon its last seen round
	if expirationParams == nil {
		expirationParams = s.expirationParams()
	}
```

We however us a helper `pmTicketParams` on the broadcaster to convert the received `net.TicketParams` to `pm.TicketParams`

This helper initialises `ticketParams.ExpirationParams` 

```		
ExpirationParams: &pm.TicketExpirationParams{
			CreationRound:          params.ExpirationParams.GetCreationRound(),
			CreationRoundBlockHash: ethcommon.BytesToHash(params.ExpirationParams.GetCreationRoundBlockHash()),
		},
```

So we actually have to check for zero values on `expirationParams` fields and not the encapsulating struct itself 

**How did you test each of these updates (required)**
Adjusted & ran unit tests
Ran Labrador on rinkeby

**Does this pull request close any open issues?**

**Checklist:**
- [ ] README and other documentation updated
- [x] Node runs in OSX and devenv
- [x] All tests in `./test.sh` pass

### CODE REVIEW

File: pm/sender.go [hunk 1]
```diff
@@ -113,7 +114,7 @@ func (s *sender) CreateTicketBatch(sessionID string, size int) (*TicketBatch, er
 	// Ensure backwards compatbility
 	// If no expirationParams are included by O
 	// B sets the values based upon its last seen round
-	if expirationParams == nil {
+	if expirationParams == nil || (expirationParams.CreationRound == 0 && expirationParams.CreationRoundBlockHash == ethcommon.Hash{}) {
```
Comment (yondonfu) [inline pm/sender.go]:
```golang
expirationParams == nil || expirationParams.CreationRound == 0 || expirationParams.CreationRoundBlockHash == ethcommon.Hash{}
```

Should the conditional be the above instead? If either field in `expirationParams` is the zero value the sender would set the expiration params on its own.

File: pm/sender_test.go [hunk 2]
```diff
@@ -297,6 +296,7 @@ func TestCreateTicketBatch_UsesSessionParamsInBatch(t *testing.T) {
 		RecipientRandHash: recipientRandHash,
 		ExpirationBlock:   big.NewInt(1),
 		PricePerPixel:     big.NewRat(1, 1),
+		ExpirationParams:  &TicketExpirationParams{},
```
Comment (yondonfu) [inline pm/sender_test.go:299]: Should we have a test that checks for a nil `ExpirationParams` as well as the new conditions?

File: pm/sender.go (hunk 1 above)
Comment (kyriediculous) [reply pm/sender.go]: Yes although both values are always used in tandem so it's never either.

We _could_ even skip the `expirationParams == nil` check since it should never be nil (due to the usage of `pmticketparams`

File: pm/sender_test.go (hunk 2 above)
Comment (kyriediculous) [reply pm/sender_test.go:299]: This should actually never be nil in the real case due to the usage of `pmTicketParams()` but I can add a test (see comment above though about the nil check)

File: pm/sender.go (hunk 1 above)
Comment (kyriediculous) [reply pm/sender.go]: fixed in c3bcc39f

File: pm/sender_test.go (hunk 2 above)
Comment (kyriediculous) [reply pm/sender_test.go:299]: fixed in c3bcc39f

**Comment by kyriediculous** [comment]
CI should pass once the rebase includes #1438

**Review by yondonfu — COMMENTED** [review]
Changes look good - let's rebase

**Comment by kyriediculous** [comment]
rebased !

**Review by yondonfu — APPROVED** [review]
LGTM 🚢
